src/CruxComponent.tsx

```diff
@@ -895,7 +895,9 @@ class CruxComponentCreator {
                                 }
 
                                 return <div key={index}>
-                                    <input key={index} type="text" value={datum}
+                                    <input key={index} 
+                                        type={this.props.field.iterabletype ? (this.props.field.iterabletype.type === "number" ? "number" : "text") : "text"} 
```

Comment:
Instead of this check you can add one more input element with type="number", so that in future it would be easy to change number specific things like min max.

Reply:
I have added new input element for type="number"